zend_abstract_interface/config/config_decode.c

```diff
@@ -131,31 +131,29 @@ static bool zai_config_decode_map(zai_str value, zval *decoded_value, bool persi
                         
                         while (*data == ' ' || *data == '\t' || *data == '\n') data++;
 
-                        if (*data == ',' || *data == '\0') {
-                            value_start = value_end = NULL;
+                        if (*data == ',' || !*data) {
+                            value_end = NULL;
                         } else {
-                            value_start = data;
-                            value_end = value_start;
-
-                            while (*data && *data != ',') {
+                            value_start = value_end = data;
+                            do {
                                 if (*data != ' ' && *data != '\t' && *data != '\n') {
                                     value_end = data;
                                 }
                                 data++;
-                            }
+                            } while (*data && *data != ',');
                         }
 
                         if (key_end && key_start) {
                             size_t key_len = key_end - key_start + 1;
-                            size_t value_len = (value_end && value_start) ? (value_end - value_start + 1) : 0;
 
                             zend_string *key = zend_string_init(key_start, key_len, persistent);
                             if (lowercase) {
                                 zend_str_tolower(ZSTR_VAL(key), ZSTR_LEN(key));
                             }
 
                             zval val;
-                            if (value_len > 0) {
+                            if (value_end) {
+                                size_t value_len = value_end ? (value_end - value_start + 1) : 0;
```

Comment:
value_end is always true here :-)

Reply:
Ooops, my bad, updated. https://github.com/DataDog/dd-trace-php/pull/2922/commits/2f104b2484299e5d02cf979a9a3c3ae39bdf2ee6